jax/_src/interpreters/mlir.py

```diff
@@ -689,6 +691,9 @@ class LoweringRuleContext:
   # module_context.shape_poly_state.dim_vars
   dim_var_values: Sequence[ir.Value] = ()
   compute_type: str | None = None
+  # Override module_context.platforms if not None. Used during multi-platform
+  # lowering, when in a scope with a subset of the module_context.platforms.
+  platforms: Sequence[str] | None = None
```

Comment:
It could be useful, but probably not much, since there is a single place in the code where platforms is set (inside the `lower_per_platform`. But it would mean adding an `__init__` which is annoying.